src/requests/cache/sorted_set/sorted_set_put_elements.rs

```diff
@@ -61,24 +148,18 @@ impl<S: IntoBytes, E: IntoBytes> SortedSetPutElementsRequest<S, E> {
 
     /// Set the time-to-live for the collection.
     pub fn with_ttl(mut self, collection_ttl: CollectionTtl) -> Self {
+        // TODO don't we want this to return a copy as opposed to mutate self?
```

Comment:
I noticed this method was called `with_ttl`, which makes it sound like a builder (hence copy constructor semantics), but it is instead mutating `self`. Was this intended or is it desired?

Reply:
Definitely seems like it'd be more consistent with the other SDKs to make this return a copy instead

Reply:
there isn't a precedent for this in any of the SDKs except for the golang one, because we don't expose request objects in any other sdks.

I don't really imagine that users are going to be keeping around a request object that they want to use as a template and continue to create clones of it via the copy constructor, and if they do want to do that, rust provides `.clone`. (side note, we should make sure all of our request objects derive clone).

rust users in general are more concerned with not wasting memory or cpu cycles by copying things that don't need to be copied, so my inclination is that we should err on the side of not wasting the memory to make the copies.

I brought up the `with_` question in sprint planning because I have had this same conversation with @nand4011 a few times now. I think it comes down to:

1. Will rust users interpret the `with_` prefix to imply a copy constructor? Or am I (slash are we) projecting that based on our work in our other SDKs?
2. If we believe that people will interpret this as a copy constructor, then should we remove the prefix? (https://github.com/momentohq/dev-eco-issue-tracker/issues/738 ) That will come at the cost of our properties/options no longer being grouped together in the autocomplete hints in the IDEs. I tried that out and it feels weird. That is why I ended up coming back around to the opinion that we should keep the prefix.

I don't think these should be copy constructors, especially for the request objects. I'm open to removing the prefix or coming up with a different one.

Thoughts?

Reply:
Checking out some popular crates that use the builder pattern:
- to mutate a field, call the method the same as the field name
- to set or unset a boolean, can get creative with verbs

[Tonic builder](https://docs.rs/tonic-build/latest/tonic_build/)
[CSV writer builder](https://docs.rs/csv/latest/csv/struct.WriterBuilder.html)

These do have the word "builder" in the name. So maybe that helps clarify expectations too. At a minimum I vote for removing `with_`

Reply:
but where are you getting the assumption that `with_` implies copy constructor? that is the assumption I'm trying to challenge in my own brain.

Also - are you unconcerned about the fact that our properties won't be grouped together in autocomplete?

Reply:
> but where are you getting the assumption that with_ implies copy constructor? that is the assumption I'm trying to challenge in my own brain.

By our consistent usage of it like this in every other SDK. Just poking around at some other crates it seems the builder is a common pattern and they default to not use prefixes.

> Also - are you unconcerned about the fact that our properties won't be grouped together in autocomplete?
They are in vscode?

![image](https://github.com/momentohq/client-sdk-rust/assets/3690240/acc20625-6491-421a-aff4-f1721d30510b)